src/backend/executor/nodeHashjoin.c

```diff
@@ -2153,3 +2170,230 @@ ExecHashJoinInitializeWorker(HashJoinState *state,
 		ExecSetExecProcNode(&state->js.ps, ExecParallelHashJoin);
 	}
 }
+
+/*
+ * Find "inner var = outer var" in hj->hashclauses and create runtime filter
+ * for it.
+ */
+void
+CreateRuntimeFilter(HashJoinState* hjstate)
+{
+	ListCell *lc;
+	AttrNumber lattno, rattno;
+
+	/*
+	 * Only applicatable for inner, right and semi join,
+	 */
+	JoinType jointype = hjstate->js.jointype;
+	if (jointype != JOIN_INNER
+		&& jointype != JOIN_RIGHT
+		&& jointype != JOIN_SEMI
+	   )
+		return;
+
+	HashState *hstate = castNode(HashState, innerPlanState(hjstate));
+	hstate->filters = NIL;
+
+	/*
+	 * check and initialize the runtime filter for all hash conds in
+	 * hj->hashclauses
+	 */
+	HashJoin *hj = castNode(HashJoin, hjstate->js.ps.plan);
+	foreach (lc, hj->hashclauses)
+	{
+		Expr *expr = (Expr *)lfirst(lc);
+
+		if (!IsEqualOp(expr))
+			continue;
+
+		lattno = -1;
+		rattno = -1;
+		if (!CheckEqualArgs(expr, &lattno, &rattno))
+			continue;
+
+		if (lattno < 1 || rattno < 1)
+			continue;
+
+		PlanState *target = FindTargetAttr(hjstate, lattno, &lattno);
+		if (lattno == -1 || target == NULL || IsA(target, HashJoinState))
+			continue;
+		Assert(IsA(target, SeqScanState));
+
+		AttrFilter *af = CreateAttrFilter(target, lattno, rattno,
+										  hstate->ps.plan->plan_rows);
+		if (af->bf)
+			hstate->filters = lappend(hstate->filters, af);
+		else
+			pfree(af);
+	}
+}
+
+static bool
+IsEqualOp(Expr *expr)
+{
+	if (!IsA(expr, OpExpr) && !IsA(expr, FuncExpr))
+		return false;
+
+	Oid funcid = InvalidOid;
+	if (IsA(expr, OpExpr))
+		funcid = ((OpExpr *)expr)->opfuncid;
+	if (IsA(expr, FuncExpr))
```

Comment:
`else if` and returns for `else` branch?

Reply:
fix it, change to` if ... else if ... else ...`